src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,179 @@
+        lambda_base::Int64, 
+        iter::Int64
+    ) 
+
+Function that updates the moving average tracking statistic. 
+
+# Inputs
+- `log::LSLogFullMA`, the moving average log structure.
+- `res::Union{AbstractVector, Real}`, the sketched residual for the current iteration. 
+- `lambda_base::Int64`, which lambda, between lambda1 and lambda2, is currently being used.
+- `iter::Int64`, the current iteration.
+
+# Outputs
+Updates the log datatype and does not explicitly return anything.
+"""
+function update_ma!(log::LSLogFullMA, res::Union{AbstractVector, Real}, lambda_base::Int64, iter::Int64)
+    # Variable to store the sum of the terms for rho
+    accum = 0
+    # Variable to store the sum of the terms for iota 
+    accum2 = 0
+    ma_info = log.ma_info
+    ma_info.idx = ma_info.idx < ma_info.lambda2 && iter != 0 ? ma_info.idx + 1 : 1
+    ma_info.res_window[ma_info.idx] = res
+    #Check if entire storage buffer can be used
+    if ma_info.lambda == ma_info.lambda2 
+        # Compute the moving average
+        for i in 1:ma_info.lambda2
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+        
+        if mod(iter, log.collection_rate) == 0 || iter == 0
+            push!(log.lambda_hist, ma_info.lambda)
+            push!(log.resid_hist, accum / ma_info.lambda) 
+            push!(log.iota_hist, accum2 / ma_info.lambda) 
+        end
+
+    else
+        # Consider the case when lambda <= lambda1 or  lambda1 < lambda < lambda2
+        diff = ma_info.idx - ma_info.lambda
+        # Because the storage of the residual is based dependent on lambda2 and 
+        # we want to sum only the previous lamdda terms we could have a situation
+        # where we want the first `idx` terms of the buffer and the last `diff`
+        # terms of the buffer. Doing this requires two loops
+        # If `diff` is negative there idx is not far enough into the buffer and
+        # two sums will be needed
+        startp1 = diff < 0 ? 1 : (diff + 1)
+        
+        # Assuming that the width of the buffer is lambda2 
+        startp2 = diff < 0 ? ma_info.lambda2 + diff + 1 : 2 
+        endp2 = diff < 0 ? ma_info.lambda2 : 1
+
+        # Compute the moving average two loop setup required when lambda < lambda2
+        for i in startp1:ma_info.idx
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+
+        for i in startp2:endp2
+            accum += ma_info.res_window[i]
+            accum2 += ma_info.res_window[i]^2
+        end
+
+        #Update the log variable with the information for this update
+        if mod(iter, log.collection_rate) == 0 || iter == 0
+            push!(log.lambda_hist, ma_info.lambda)
+            push!(log.resid_hist, accum / ma_info.lambda) 
+            push!(log.iota_hist, accum2 / ma_info.lambda) 
+        end
+        
+        ma_info.lambda += ma_info.lambda < lambda_base ? 1 : 0
+    end
+
+end
```

Comment:
the `update_ma!` function should be placed in a utilities/helper folder and accept both inputs as arguments (assuming that the code is identical)

It is also probably better to move `MAInfo` to this directory as well.